haskell/cabal.bzl

```diff
@@ -1017,11 +1020,11 @@ _STACK_BINDISTS = \
 
 def _stack_version_check(repository_ctx, stack_cmd):
     """Returns False if version not recent enough."""
-    exec_result = _execute_or_fail_loudly(repository_ctx, [stack_cmd, "--numeric-version"])
-
-    stack_major_version = int(exec_result.stdout.split(".")[0])
-    stack_minor_version = int(exec_result.stdout.split(".")[1])
-    return stack_major_version >= 2 and stack_minor_version >= 3
+    exec_result = repository_ctx.execute([stack_cmd, "--numeric-version"])
+    if exec_result.return_code != 0:
+        return False, "not"  # HACK: use as "Stack {} found".format(version)
+    version = tuple([int(x) for x in exec_result.stdout.split(".")[:2]])
+    return version >= _STACK_MIN_VERSION, exec_result.stdout.strip()
```

Comment:
IIUC all use-sites construct the same message based on this output. So, maybe an easy way to avoid the hack and also avoid some duplication is to construct the full message here and return it.